activiti-services/activiti-services-audit/activiti-services-audit-jpa/src/test/java/org/activiti/services/history/HistoryServiceIT.java

```diff
@@ -0,0 +1,102 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.stream.binder.test.junit.rabbit.RabbitTestSupport;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource("classpath:application-producer-test.properties")
+@DirtiesContext
+public class HistoryServiceIT {
+
+    private static final String relativeQueryEndpoint = "/audit/events";
+
+    @ClassRule
+    public static RabbitTestSupport rabbitTestSupport = new RabbitTestSupport();
+
+    @Autowired
+    private EventsRestTemplate eventsRestTemplate;
+
+    @Autowired
+    private EventsRepository repository;
+
+    @Autowired
+    private MyProducer producer;
+
+    @Before
+    public void setUp() throws Exception {
+        repository.deleteAll();
+    }
+
+    @Test
+    public void findAllShouldReturnAllAvailableEvents() throws Exception {
+        //given
+        producer.send(new MockProcessEngineEvent(System.currentTimeMillis(),
+                                                 "ActivityStartedEvent",
+                                                 "2",
+                                                 "3",
+                                                 "4"));
+        producer.send(new MockProcessEngineEvent(System.currentTimeMillis(),
+                                                 "ActivityCompletedEvent",
+                                                 "11",
+                                                 "23",
+                                                 "42"));
+        waitForMessage();
+
+        //when
+        ResponseEntity<PagedResources<ProcessEngineEventEntity>> eventsPagedResources = eventsRestTemplate.executeFindAll();
+
+        //then
+        assertThat(eventsPagedResources.getBody().getContent())
+                .extracting(
+                        ProcessEngineEventEntity::getEventType,
+                        ProcessEngineEventEntity::getExecutionId,
+                        ProcessEngineEventEntity::getProcessDefinitionId,
+                        ProcessEngineEventEntity::getProcessInstanceId)
+                .contains(
+                        tuple("ActivityStartedEvent",
+                              "2",
+                              "3",
+                              "4"),
+                        tuple("ActivityCompletedEvent",
+                              "11",
+                              "23",
+                              "42"));
+    }
+
+    private void waitForMessage() throws InterruptedException {
```

Comment:
@erdemedeiros I'm thinking about this.. This is simulating the network latency for a fire and forget use case (the engine emitting events not worried about if someone is going to consume it). So not sure if we can do much to improve this. Unless we find a way to ack the message.